script.js

```diff
@@ -0,0 +1,70 @@
+
+const searchButton = document.getElementById('searchButton');
+
+searchButton.addEventListener('click', () => {
+    document.getElementById("tableResults").innerHTML = '';
+    searchResults()
+    addSpinner();
+});
+
+async function searchResults() {
+    const symbol = document.getElementById('searchTerm').value;
+    const url = "https://stock-exchange-dot-full-stack-course-services.ew.r.appspot.com/api/v3/search?query="+ symbol + "&limit=10&exchange=NASDAQ"
+    const response = await fetch(url)
+    const result = await response.json();
+    
+    stopSpinner();
+    
+    const tableResults = document.getElementById("tableResults");
+    let searchResults = result;
+    try {
+        for (let i = 0; i < searchResults.length; i++) {
+            let currentResult = searchResults[i];
+            tableResults.innerHTML += ('<img src="https://financialmodelingprep.com/images-New-jpg/' + currentResult.symbol + '.jpg"> ' + "<a href='company.html?symbol=" + currentResult.symbol +"'>"  + currentResult.name  + ' </a>' + '(' + currentResult.symbol + ')');
+
+
+            
+            const url2 = "https://stock-exchange-dot-full-stack-course-services.ew.r.appspot.com/api/v3/company/profile/" + currentResult.symbol;
+            const response2 = await fetch(url2);
```

Comment:
here you should use promise.all() in order to send all the HTTP reuqests in the same time, instead of doing it synchronously.